### DISCUSSION TITLE
Wiki Additions

### DISCUSSION THREAD

**Discussion by J-The-Fox** [J-The-Fox/FRC-Team-5098#3] [Ideas]
Anything that should be added to the Wiki that may have been forgotten or that you think would be helpful?

**Comment by Dreadfall** [comment]
Probably need to specify a difference between functions and methods at some point
just putting this here to remember for later

**Reply by J-The-Fox** [reply]
**Here is a bit more on that topic:**

> From this [Stack Overflow question](https://stackoverflow.com/questions/155609/whats-the-difference-between-a-method-and-a-function)

---

A **function** is a piece of code that is called by name. It can be passed data to operate on (i.e. the parameters) and can optionally return data (the return value). All data that is passed to a function is explicitly passed.

A **method** is a piece of code that is called by a name that is associated with an object. In most respects it is identical to a function except for two key differences:

- A method is implicitly passed the object on which it was called.
- A method is able to operate on data that is contained within the class (remembering that an object is an instance of a class - the class is the definition, the object is an instance of that data).

(this is a simplified explanation, ignoring issues of [scope](http://en.wikipedia.org/wiki/Scope_(computer_science)) etc.)

---

Hopefully this can help in separating the two although they are both quite similar
